src/Microsoft.Azure.WebJobs.Host/TypeUtility.cs

```diff
@@ -98,6 +98,68 @@ internal static Type UnwrapTaskType(Type type)
             return null;
         }
 
+        // $$$
+        internal static TAttribute GetAttr<TAttribute>(ParameterInfo parameter) where TAttribute : Attribute
+        {
+            TAttribute attributeSource = parameter.GetCustomAttribute<TAttribute>(inherit: false);
+
+            // Walk hierachy.
+            // It's possible attribute is not on the parameter but is still on the something 
+            // higher up in the hiearchy
+            ICustomAttributeProvider provider = parameter;
+
+            while (provider != null)
+            {
+                var attrs = provider.GetCustomAttributes(typeof(Attribute), false);
+                foreach (var attr in attrs)
+                {
+                    if (attributeSource == null)
```

Comment:
It's not clear to me at all what this code is doing. Some good code comments in this method would go a long way